src/games/strategy/engine/framework/GameRunner.java

```diff
@@ -222,212 +226,168 @@ private static void handleCommandLineArgs(final String[] args) {
+          printUsage = true;
         }
       }
-    });
-  }
 
-  public static void setupLogging() {
-    // setup logging to read our logging.properties
-    try {
-      LogManager.getLogManager().readConfiguration(ClassLoader.getSystemResourceAsStream("logging.properties"));
-    } catch (final Exception e) {
-      ClientLogger.logQuietly(e);
-    }
-  }
+      if (printUsage || usagePrinted) {
+        usage(gameMode);
+        System.exit(-1);
+      }
 
-  private static String getDefaultLookAndFeel() {
-    final Preferences pref = Preferences.userNodeForPackage(GameRunner.class);
-    // substance 7.x
-    String defaultLookAndFeel = SubstanceGraphiteLookAndFeel.class.getName();
-    // macs are already beautiful
-    if (isMac()) {
-      defaultLookAndFeel = UIManager.getSystemLookAndFeelClassName();
-    }
-    final String userDefault = pref.get(LOOK_AND_FEEL_PREF, defaultLookAndFeel);
-    final List<String> availableSkins = TripleAMenuBar.getLookAndFeelAvailableList();
-    if (!availableSkins.contains(userDefault)) {
-      if (!availableSkins.contains(defaultLookAndFeel)) {
-        return UIManager.getSystemLookAndFeelClassName();
+    } else {
+
+
+      final String version = System.getProperty(TRIPLEA_ENGINE_VERSION_BIN);
+      if (version != null && version.length() > 0) {
+        final Version testVersion;
+        try {
+          testVersion = new Version(version);
+          // if successful we don't do anything
+          System.out.println(TRIPLEA_ENGINE_VERSION_BIN + ":" + version);
+          if (!ClientContext.engineVersion().getVersion().equals(testVersion, false)) {
+            System.out.println("Current Engine version in use: " + ClientContext.engineVersion());
+          }
+        } catch (final Exception e) {
+          System.getProperties().setProperty(TRIPLEA_ENGINE_VERSION_BIN, ClientContext.engineVersion().toString());
+          System.out.println(TRIPLEA_ENGINE_VERSION_BIN + ":" + ClientContext.engineVersion());
+        }
+      } else {
+        System.getProperties().setProperty(TRIPLEA_ENGINE_VERSION_BIN, ClientContext.engineVersion().toString());
+        System.out.println(TRIPLEA_ENGINE_VERSION_BIN + ":" + ClientContext.engineVersion());
       }
-      setDefaultLookAndFeel(defaultLookAndFeel);
-      return defaultLookAndFeel;
     }
-    return userDefault;
   }
 
-  public static void setDefaultLookAndFeel(final String lookAndFeelClassName) {
-    final Preferences pref = Preferences.userNodeForPackage(GameRunner.class);
-    pref.put(LOOK_AND_FEEL_PREF, lookAndFeelClassName);
-    try {
-      pref.sync();
-    } catch (final BackingStoreException e) {
-      ClientLogger.logQuietly(e);
+  private static boolean argsStartsWithPropertykey(String firstArg) {
+    boolean startsWithPropertyKey = false;
+    for (final String arg : COMMAND_LINE_ARGS) {
+      if (firstArg.startsWith(arg)) {
+        startsWithPropertyKey = true;
+        break;
```

Comment:
Addressed with 5e47d1a (see commit comment there pls)
Why change it to a foreach when we can just delete the method :boom: